community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/planner/logical/steps/pickBestPlanUsingHintsAndCost.scala

```diff
@@ -26,7 +26,7 @@ object pickBestPlanUsingHintsAndCost extends LogicalPlanningFunction0[CandidateS
   private final val VERBOSE = false
 
   override def apply(implicit context: LogicalPlanningContext): CandidateSelector = new CandidateSelector {
-    override def apply[X](projector: (X) => LogicalPlan, input: Iterable[X]): Option[X] = {
+    override def apply[X](projector: (X) => LogicalPlan, input: Iterator[X]): Option[X] = {
```

Comment:
I like the idea to have an implicit for fixing this, we need to try though if that confuses scala. 

On seq: This is about {expand|join}solverStep which both are written as big for comprehensions.
It is hard to package that up as an iterable without creating an intermediary seq but now that I think about it, maybe not :) I'll give it a try that was good input, thanks @pontusmelke